### ISSUE TITLE
Analysis failure on boxing of `UNARY_MINUS` expression

### ISSUE THREAD

**Issue by Stephan202** [uber/NullAway#39]
Consider the following code:
```java
final class Container {
    private void sender() {
        int someValue = 0;
        receiver(-someValue);
    }

    private void receiver(Integer i) {
        /* NOP */
    }
}
```

Analyzing this code with NullAway 0.1.6 inside Error Prone 2.1.1 yields:
```
java.lang.RuntimeException: whoops, better handle UNARY_MINUS -someValue
 	at com.uber.nullaway.NullAway.mayBeNullExpr(NullAway.java:1075)
 	at com.uber.nullaway.NullAway.handleInvocation(NullAway.java:644)
 	at com.uber.nullaway.NullAway.matchMethodInvocation(NullAway.java:262)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitMethodInvocation(ErrorProneScanner.java:907)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitMethodInvocation(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1644)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
 	at com.sun.source.util.TreeScanner.visitExpressionStatement(TreeScanner.java:433)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitExpressionStatement(ErrorProneScanner.java:713)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitExpressionStatement(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1454)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
 	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
 	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
 	at com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:248)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:530)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitBlock(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1026)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
 	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
 	at com.sun.source.util.TreeScanner.visitMethod(TreeScanner.java:206)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:898)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:898)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
 	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
 	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
 	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
 	at com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:590)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:82)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:42)
 	at com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
 	at com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
 	at com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:144)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:605)
 	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:146)
 	at com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:591)
 	at com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
 	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:64)
 	at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:41)
 	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:145)
 	at com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:120)
 	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1425)
 	at com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1374)
 	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:973)
 	at com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:100)
 	at com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:142)
 	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:96)
 	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:90)
 	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:137)
 	at com.google.errorprone.BaseErrorProneCompiler.run(BaseErrorProneCompiler.java:108)
 	at com.google.errorprone.ErrorProneCompiler.run(ErrorProneCompiler.java:118)
 	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne$CompilerInvoker.compile(JavacCompilerWithErrorProne.java:222)
 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 	at java.lang.reflect.Method.invoke(Method.java:498)
 	at org.codehaus.plexus.compiler.javac.errorprone.JavacCompilerWithErrorProne.performCompile(JavacCompilerWithErrorProne.java:91)
 	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:1075)
 	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:168)
 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:200)
 	at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:196)
 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 	at java.lang.Thread.run(Thread.java:748)
```

The issue is also triggered if `someValue` is replaced with another `int` or `Integer`-typed non-constant expression such as a method invocation. The issue is not triggered if `receiver`'s signature is updated such that the method accepts a primitive `int`.

**Comment by msridhar** [comment]
Oof, sorry about this one 🙁 A fix is up.  We can cut a new release after it lands if that would be helpful

**Comment by Stephan202** [comment]
@msridhar, thanks a lot for the super-fast turnaround time :+1: . A new release would be nice so that I can continue evaluating NullAway (this was the only failure when I ran the plugin against a project with ~280k LOC), but it's totally fine if you prefer to wait until the next "regular" release moment. I assume it's also almost weekend in the US ;)

**Comment by msridhar** [comment]
The fix has landed on master.  It shouldn't be *too* hard for you to cut a local SNAPSHOT release to test further (just do `./gradlew install` and then reference version `0.1.7-SNAPSHOT` from your other project).  Probably we will cut a new release early next week.  Thanks again for the report!

**Comment by Stephan202** [comment]
Thanks! I can confirm that with this fix our build passes (because I disabled `-Werror`, of course ;) ).

Instead of a locally built artifact I managed to successfully pull in the changes using the [JitPack.io](http://jitpack.io/) Maven coordinates `com.github.uber:nullaway:7b7fbb467a56bbbd5d7c194fd99cd7ff671734d4`. ([Build log](https://jitpack.io/com/github/uber/NullAway/7b7fbb467a56bbbd5d7c194fd99cd7ff671734d4/build.log).)
